### PR TITLE
Timeseries calibration and combination logic

### PR DESCRIPTION
**Pull request by liviajakob**
Created new module timeseries_combination_helpers.py , which contains a new logic to calibrate trends with a higher resolution timeseries. For more advanced combination rather than simple averaging.

Note that there is a function called combine_timeseries (now renamed combine_timeseries_imbie) in util.timeseries_helpers which is the simple resampling and averaging logic taken from IMBIE. This can ultimately be removed, but will be kept for now to do a comparison.

### CODE REVIEW

File: tests/util/test_timeseries_combination_helpers.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    expected_correction = 2 / 3
+    assert np.array_equal(calibrated_ts[1], np.array(calibration_timeseries["changes"]) + expected_correction)
+
+    # check distance matrix
+    assert np.array_equal(distance_matrix[0], np.array([1., 1., 2., 3., 4.]))
+    assert np.array_equal(distance_matrix[1], np.array([3., 2., 1., 1., 1.]))
+
+
+def test_combine_calibrated_timeseries():
+    calibrated_series = np.array([[1., 2., 3., 2., 1.],
```
Comment (alex-ew) [inline tests/util/test_timeseries_combination_helpers.py:56]: Given they're repeated below, it's cleaner to extract calibrated_series and distance_matrix as test fixtures

File: glambie/util/timeseries_combination_helpers.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
+import warnings
+import numpy as np
+import pandas as pd
+
+
+def calibrate_timeseries_with_trends(trends: pd.DataFrame, calibration_timeseries: pd.DataFrame):
+    """
+    This function calibrates a higher resolution calibarion timeseries with the trends from a trend DataFrame
```
Comment (alex-ew) [inline glambie/util/timeseries_combination_helpers.py:8]: Minor typo 'calibarion'. Also, line should finish with a .

File: glambie/util/timeseries_combination_helpers.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    The 'calibration_timeseries' is adjusted over the common time period, to represent the new trend.
+    Each trend within 'trends' (i.e. each column) will produce a separate calibrated solution from the
```
Comment (alex-ew) [inline glambie/util/timeseries_combination_helpers.py:10]: This contradicts the parameters summary which says there a exactly 3 columns

**Review by alex-ew — COMMENTED** [review]
Generally looks good and well defined test cases. There are a number of small typos and some other minor suggestions. Open to discussion as always

File: glambie/util/timeseries_combination_helpers.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+                               period_start_date: float,
+                               period_end_date: float,
+                               resolution: float = 1 / 12) -> float:
+    """
+    Calculates the distance to a time period in years for distance weighting.
+
+    Parameters
+    ----------
+    date : float
+        date to calcualte the distance for in fractional years
```
Comment (alex-ew) [inline glambie/util/timeseries_combination_helpers.py:115]: calcualte, and "for in" unclear

File: glambie/util/timeseries_combination_helpers.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+    Returns
+    -------
+    Tuple(np.array[[]], np.array[[]])
+        Returns a tuple of two 2D lists
+        1.) 2D array with calibrated series
+            will contain the dimensions of: (a) the trends dataframe length and (b) length of calibarion_timeseries
+        2.) 2D array with a distance matrix for inverse distance calculations
+            will contain the same dimensions as (1)
+    """
```
[the remark this answers is not in the document]
Comment (liviajakob) [reply glambie/util/timeseries_combination_helpers.py:39]: fixed

[the remark this answers is not in the document]
Comment (liviajakob) [reply glambie/util/timeseries_combination_helpers.py:39]: fixed

[the remark this answers is not in the document]
Comment (alex-ew) [reply glambie/util/timeseries_combination_helpers.py:39]: Thanks for the extra context. I see the extra mention in the doc string. Cheers

File: glambie/util/timeseries_combination_helpers.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+                               resolution: float = 1 / 12) -> float:
+    """
+    Calculates the distance to a time period in years for distance weighting.
+
+    Parameters
+    ----------
+    date : float
+        date to calcualte the distance for in fractional years
+        date is assumed to be at the start within a time series (and not the middle of a timestep)
+        i.e. date is a start_date
```
[the remark this answers is not in the document]
Comment (alex-ew) [reply glambie/util/timeseries_combination_helpers.py:120]: Could maybe call it trend_start_date. I'll resolve the comment for now. This is a fair reply

**Review by alex-ew — APPROVED** [review]
Looks good - thanks for the changes

**Pull request merged**